src/styles/animations.js

```diff
@@ -0,0 +1,69 @@
+var AutoPrefix = require('./auto-prefix');
+
+var createdAnimations = {};
+
+function createSheet(){
+
+	var styleElement = document.createElement("style");
+	document.head.appendChild(styleElement);
+	var sheet = styleElement.sheet;
+
+	return sheet;
+}
+
+function addRule(sheet, selector){
+	var size = (sheet.cssRules || sheet.rules).length;
+	if("insertRule" in sheet) {
+		sheet.insertRule(selector + " { }", size);
+	}
+	else if("appendRule" in sheet){
+		sheet.appendRule(selector + " { }");
+	}
+	else if("addRule" in sheet) {
+		sheet.addRule(selector, " {} ", size);
+	}
+}
+ 
+function setStyles(rule, obj){
+	for(var prop in obj){
+		rule.style[prop] = obj[prop];
+	}
+}
+
+module.exports = {
+
+	 create: function (name, frames){
+
+	 	if(name in createdAnimations){
+	 		return name;
+	 	}
+	 	
+	 	createdAnimations[name] = true;
+
+		var sheet = createSheet();
+		 
+		var keyframeTags = [ "@keyframes ", "@-webkit-keyframes "];
```

Comment:
My implementation relied on transitionend event. Worked pretty well for the linear progress indicator but fell apart when applied to circular. The circular progress was very jumpy; the animations smoothed it out.

I much prefer using animations when I need to repeat something.You can accomplish this using the transitionend event but 'resetting' the element can be tricky. For the circle, you could try rotate 360 and on transitionend set the rotation to 0 outside of the render method and then get the height (I think height will 'redraw' the element).